lib/rdkafka/bindings.rb

```diff
@@ -246,14 +246,21 @@ class TopicPartitionList < FFI::Struct
     attach_function :rd_kafka_conf_set_dr_msg_cb, [:pointer, :delivery_cb], :void
 
     # Partitioner
-    attach_function :rd_kafka_msg_partitioner_consistent_random, [:pointer, :pointer, :size_t, :int32, :pointer, :pointer], :int32
+    PARTITIONERS = %w(random consistent consistent_random murmur2 murmur2_random fnv1a fnv1a_random).each_with_object({}) do |name, hsh|
+      method_name = "rd_kafka_msg_partitioner_#{name}".to_sym
+      attach_function method_name, [:pointer, :pointer, :size_t, :int32, :pointer, :pointer], :int32
+      hsh[name] = method_name
+    end
 
-    def self.partitioner(str, partition_count)
+    def self.partitioner(str, partition_count, partitioner_name = "consistent_random")
```

Comment:
Maybe maken this `nil` by default and skip setting it? Then the default behaviour stays the same.

Reply:
I'm sorry but I'm afraid I don't understand your intention. I set the default value "consistent_random" for the users who call this method directly like `Rdkafka::Bindings.partitioner(str)`, and I think the default behavior, including such a case, hasn't changed.
Which do you suggest?

```ruby
def self.partitioner(str, partition_count, partitioner_name = nil)
  ...    
end
```

or

```ruby
def self.partitioner(str, partition_count, partitioner_name)
  ...    
end
```